// hemera@0.0.5/lib/hemera.js
var url = require('url');
var request = require('request');
var base_path = 'http://api.douban.com/v2/movie/';


module.exports = Hemera;


function Hemera() {}


Hemera.prototype.getMovies= function(url, callback) {
    
    request(url, function(err, res, body) {
        if (err || res.statusCode !== 200) {
            throw new Error('get movies request failed ' + err + res.statusCode);
        } else {
            body = JSON.parse(body);
            callback(body);
        }
    });
};


Hemera.prototype.getMovieById = function(id, callback) {
    
    request(base_path + 'subject/' + id, function(err, res, body) {
        if (err || res.statusCode !== 200) {
            throw new Error('get movie by id request failed ' + err + res.statusCode);
        } else {
            callback(res, body);
        } 
    });
};


// 影人条目信息
Hemera.prototype.getCelebrityById = function(id, callback) {
    
    request(base_path + 'celebrity/' + id, function(err, res, body) {
        if (err || res.statusCode !== 200) {
            throw new Error('get celebrity by id request failed ' + err + res.statusCode);
        } else {
            body = JSON.parse(body);
            callback(body);
        }
    });
};


// 电影评论
Hemera.prototype.getReviewsById = function(id, callback) {

    request(base_path + 'subject' + id + 'reviews', function() {
        if (err || res.statusCode !== 200) {
            throw new Error('get reviews by id request failed ' + err + res.statusCode);
        } else {
            body = JSON.parse(body);
            callback(body);
        }
        
    });
};


// 影人作品
Hemera.prototype.getWorksById = function(id, callback) {
    
    request(base_path + 'celebrity/' + id + 'works', function(err, res, body) {
        if (err || res.statusCode !== 200) {
            throw new Error('get works by id request failed ' + err + res.statusCode);
        } else {
            body = JSON.parse(body);
            callback(body);
        }
    });
};


// 电影条目搜索
Hemera.prototype.search = function(text, callback) {

    var search_path = url.parse('http://api.douban.com/v2/movie/search/');
    
    request({uri: 'http://api.douban.com/v2/movie/search', qs: text}, function(err, res, body) {
        if (err || res.statusCode !== 200) {
            throw new Error('search request failed ' + err + res.statusCode);
        } else {
            body = JSON.parse(body);
            callback(body);
        }
    });
};
